django/thunderstore/security/utils.py

```diff
@@ -0,0 +1,140 @@
+import base64
+import binascii
+import json
+
+import requests
+from cryptography.hazmat.backends.openssl.backend import backend
+from cryptography.hazmat.primitives.asymmetric import ec
+from six import b, text_type
+
+
+class DERStringDecoder(object):
+    def __init__(self, initial_bytes):
+        self._buffer = initial_bytes
+        self._index = 0
+
+    def remaining_data(self):
+        return len(self._buffer) - self._index
+
+    def read(self, length):
+        new_index = self._index + length
+        if new_index > len(self._buffer):
+            raise ValueError(
+                "Index is over the buffer index: %d buffer length: %d"
+                % (new_index, len(self._buffer))
+            )
+        result = self._buffer[self._index : new_index]
+        self._index = new_index
+        return int(result, 16)
+
+    def decode(self):
+        comps = [str(x) for x in divmod(self.read(2), 40)]
+        v = 0
+        while self.remaining_data():
+            c = self.read(2)
+            v = v * 128 + (c & 0x7F)
+            if not (c & 0x80):
+                comps.append(str(v))
+                v = 0
+        decoded = ".".join(comps)
+        return decoded
+
+
+def unpem(pem):
+    if isinstance(pem, text_type):  # pragma: no branch
+        pem = pem.encode()
```

Comment:
since proper type hinting will make this illegal, just write the whole thing as:
```python
pem_data = pem.encode() if isinstance(pem, str) else pem
```

Reply:
Actually looking at this a bit more, why are we converting to bytes and not string? This leads to lines 47-49 having to convert all parameters into bytes when doing processing too. It would feel more natural to handle the entire thing as text, and convert to bytes only when needed as bytes.

Moreover, does the function need to be called with bytes at any point?